```
  1551 passing (2s)
  12 pending
  1 failing

  1) fake
       should reject non-Function argument:
     AssertionError: [assert.exception] Expected exception
      at Object.fail (node_modules/@sinonjs/referee/lib/create-fail.js:12:21)
      at Object.fail (node_modules/@sinonjs/referee/lib/define-assertion/index.js:57:17)
      at Object.assert (node_modules/@sinonjs/referee/lib/assertions/exception.js:44:21)
      at assertion (node_modules/@sinonjs/referee/lib/define-assertion/index.js:62:23)
      at referee.<computed>.<computed> [as exception] (node_modules/@sinonjs/referee/lib/define-assertion/index.js:111:22)
      at file:///home/runner/work/sinon/sinon/test/src/fake-test.js:94:20
      at Array.forEach (<anonymous>)
      at Context.<anonymous> (file:///home/runner/work/sinon/sinon/test/src/fake-test.js:93:18)
      at process.processImmediate (node:internal/timers:504:21)

```

--- a/src/sinon/fake.js
+++ b/src/sinon/fake.js
@@ -18,11 +18,12 @@ const { slice } = prototypes.array;
  *
  * @param {SinonFunction|undefined} [f]
  * @param {object} [context] The sinon context for callId tracking
+ * @param {boolean} hasFunctionArgument
  * @returns {SinonFunction}
  * @namespace
  */
-function fakeImpl(f, context) {
-    if (typeof f !== "undefined" && typeof f !== "function") {
+function fakeImpl(f, context, hasFunctionArgument) {
+    if (hasFunctionArgument && typeof f !== "function") {
         throw new TypeError("Expected f argument to be a Function");
     }
 
@@ -39,7 +40,7 @@ function fakeImpl(f, context) {
  * @namespace
  */
 function fake(f) {
-    return fakeImpl(f, undefined);
+    return fakeImpl(f, undefined, arguments.length > 0);
 }
 
 /**
@@ -50,7 +51,7 @@ function fake(f) {
  * @returns {SinonFunction}
  */
 fake.withContext = function (context, f) {
-    return fakeImpl(f, context);
+    return fakeImpl(f, context, arguments.length > 1);
 };
 
 /**

